packages/third-parties/src/google/gtm.tsx

```diff
@@ -0,0 +1,61 @@
+'use client'
+// TODO: Evaluate import 'client only'
+import React from 'react'
+import Script from 'next/script'
+
+declare global {
+  interface Window {
+    dataLayer?: Object[]
+    [key: string]: any
+  }
+}
+
+type GTMParams = {
+  gtmId: string
+  dataLayer: string[]
+  dataLayerName: string
+  auth: string
+  preview: string
+}
+
+let currDataLayerName = 'dataLayer'
+
+export function GoogleTagManager(props: GTMParams) {
+  const { gtmId, dataLayerName = 'dataLayer', auth, preview, dataLayer } = props
+
+  currDataLayerName = dataLayerName
```

Comment:
I don't think this pattern is edge-case free. When you render this component it will use the initial dataLayer prop to instantiate the script by id. Then afterwards any changes to the dataLayer prop will not be reflected in teh gtm script but will alter this module scoped variable.

The `sendGTMEvent` refers to the current dataLayer rather than the the one initially used instantiate the script so a wayward props update could breat gtm tracking.

If it is true that you can't hot-update the datalayer in GTM (meaning you need to just pick it at the start of the app runtime and never change it) then we should align the module global to only update when the gtm script will actually run which is just once one hte first render.

Reply:
That makes sense! I have updated the code to only modify `currDataLayerName` during initialization